### PR TITLE
[Gatsby] "https://facebook.github.io/react/" -> "https://reactjs.org/"

### PR DESCRIPTION
**Pull request by skratchdot**
Not sure this change is wanted (or too excessive).  The biggest question I had is whether it was okay to update old changelog links.  I smoke tested a few of the link changes, and everything seems to be working.

### CODE REVIEW

**Comment by skratchdot** [comment]
2 other questionable changes from this pr:
- [docs/_config.yml](https://github.com/facebook/react/pull/10970/files#diff-b09bbff3e688a10b35f7de810d65c28e)
- [docs/_layouts/default.html](https://github.com/facebook/react/pull/10970/files#diff-8c608efa00dc2076c00bd9b794310a08) (the og:url link - will it generate links with extraneous slashes?)

**Comment by skratchdot** [comment]
`og:url` are still showing undefined (but they are doing that in the currently deployed version.

```<meta data-react-helmet="true" property="og:url" content="undefined/index.md"/>```

**Comment by bvaughn** [comment]
> The biggest question I had is whether it was okay to update old changelog links.

Yeah, I think this is reasonable.

> 2 other questionable changes from this pr:
> * docs/_config.yml
> * docs/_layouts/default.html (the og:url link - will it generate links with extraneous slashes?)

Don't worry about changes to `docs/_layouts/default.html`. This is used for the old Jekyll site which the new site has replaced.

> `og:url` are still showing undefined (but they are doing that in the currently deployed version.

`og:url` does not show as undefined for me (on the live site) although it does show the incorrect file extensions (`.md` instead of `.html`). This should hopefully be fixed by #10902 (or similar).

**Comment by skratchdot** [comment]
@bvaughn - I missed a nuance in the `og:url`.  If you view the generated dom after js kicks in (and react-helmet updates the url), then I see reactjs.org.  I was using "view-source", so it was showing up as undefined.  This is probably something that can be fixed in the gatsby build (but I've never used gatsby).  I'll try to see if there's anything that can be done.  Probably not a big deal in the short term.

File: src/isomorphic/children/onlyChild.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ var invariant = require('fbjs/lib/invariant');
  * Returns the first child in a collection of children and verifies that there
  * is only one child in the collection.
  *
- * See https://facebook.github.io/react/docs/react-api.html#react.children.only
+ * See https://reactjs.org/docs/react-api.html#react.children.only
```
Comment (bvaughn) [inline src/isomorphic/children/onlyChild.js]: A couple of these anchor tags are invalid b'c of how Gatsby handles them. For instance, this one should now be: reactjs.org/docs/react-api.html#reactchildrenonly

But this is not related to this PR 😄 Just mentioning it in passing.
